catalog/spatial/registry/registry-federation-admin-service/src/main/java/org/codice/ddf/registry/federationadmin/service/FederationAdminService.java

```diff
@@ -14,42 +14,223 @@
+     *
+     * @param metacard Metacard to be stored
+     * @return Id of the created metacard
+     * @throws FederationAdminException If the metacard provided was null.
+     *                                  The CatalogFramework.create call throws IngestException or SourceUnavailableException
+     */
+    String addRegistryEntry(Metacard metacard) throws FederationAdminException;
+
+    /**
+     * Add the given metacard to the registry catalog with the provided destinations
+     *
+     * @param metacard     Metacard to be stored
+     * @param destinations Set of destinations to add
+     * @return Id of the created metacard
+     * @throws FederationAdminException If the metacard provided was null.
+     *                                  The CatalogFramework.create call throws IngestException or SourceUnavailableException
+     */
+    String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException;
+
+    /**
+     * * Deletes the registry entry from the registry catalog for each of the provided registry ids.
+     *
+     * @param registryIds The list of registry ids to be deleted.
+     * @throws FederationAdminException If the list of ids provided is empty.
+     *                                  If IngestException or Source UnavailableException was thrown by CatalogFramework.delete
+     */
+    void deleteRegistryEntriesByRegistryIds(List<String> registryIds)
+            throws FederationAdminException;
+
+    /**
+     * Deletes the registry entry from the registry catalog for each of the provided registry ids. A set of destinations is
+     * sent in the delete request
+     *
+     * @param registryIds  The list of registry ids to be deleted.
+     * @param destinations Set of destinations to add
+     * @throws FederationAdminException If the list of ids provided is empty.
+     *                                  If IngestException or Source UnavailableException was thrown by CatalogFramework.delete
+     */
+    void deleteRegistryEntriesByRegistryIds(List<String> registryIds, Set<String> destinations)
+            throws FederationAdminException;
+
+    /**
+     * * Deletes the registry entry from the registry catalog for each of the provided metacard ids.
+     *
+     * @param metacardIds The list of metacard ids to be deleted.
+     * @throws FederationAdminException If the list of ids provided is empty.
+     *                                  If IngestException or Source UnavailableException was thrown by CatalogFramework.delete
+     */
+    void deleteRegistryEntriesByMetacardIds(List<String> metacardIds)
+            throws FederationAdminException;
+
+    /**
+     * * Deletes the registry entry from the registry catalog for each of the provided metacard ids. A set of destinations is
+     * sent in the delete request
+     *
+     * @param metacardIds  The list of metacard ids to be deleted.
+     * @param destinations The set of destinations to be added
```

Comment:
Added should be deleted.  Looks like destinations is really a set of source ids. Might consider changing all destinations to sourceIds for consistency with the catalog framework API.